packages/typescript-estree/src/convert.ts

```diff
@@ -3319,23 +3319,30 @@ export class Converter {
     aliasKey: AliasKey,
     valueKey: ValueKey,
   ): Properties & Record<AliasKey, Properties[ValueKey]> {
-    let errored = false;
+    let warned = false;
 
     Object.defineProperty(node, aliasKey, {
+      configurable: true,
       get: this.options.suppressDeprecatedPropertyWarnings
         ? (): Properties[typeof valueKey] => node[valueKey]
         : (): Properties[typeof valueKey] => {
-            if (!this.options.suppressDeprecatedPropertyWarnings) {
-              if (!errored) {
-                // eslint-disable-next-line no-console
-                console.warn(
-                  `The '${aliasKey}' property is deprecated on ${node.type} nodes. Use '${valueKey}' instead. See https://typescript-eslint.io/linting/troubleshooting#the-key-property-is-deprecated-on-type-nodes-use-key-instead-warnings.`,
-                );
-                errored = true;
-              }
+            if (!warned) {
+              // eslint-disable-next-line no-console
+              console.warn(
```

Comment:
should we use nodejs's infra here?
[`process.emitWarning`](https://nodejs.org/api/process.html#processemitwarningwarning-options)

eg
```ts
process.emitWarning(
  `The '${aliasKey}' property is deprecated on ${node.type} nodes. Use '${valueKey}' instead. See https://typescript-eslint.io/linting/troubleshooting#the-key-property-is-deprecated-on-type-nodes-use-key-instead-warnings.`
	'DeprecationWarning',
)
```

Reply:
TIL, thanks!